pkg/catalogv2/git/config.go

```diff
@@ -0,0 +1,144 @@
+// If a CA bundle is provided, it converts the CA bundle from DER to PEM format, since Git requires PEM format.
+// In this case, insecureSkipTLS is set to false since a CA bundle is provided for secure communication.
+// Finally, it returns a new git object configured with these settings,
+// or an error if any step in this process fails.
+func gitForRepo(secret *corev1.Secret, namespace, name, gitURL string, insecureSkipTLS bool, caBundle []byte) (*git, error) {
+	isGitSSH, err := isGitSSH(gitURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to verify the type of URL %s: %w", gitURL, err)
+	}
+	if !isGitSSH {
+		u, err := url.Parse(gitURL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse URL %s: %w", gitURL, err)
+		}
+		if u.Scheme != "http" && u.Scheme != "https" {
+			return nil, fmt.Errorf("invalid git URL scheme %s, only http(s) and git supported", u.Scheme)
+		}
+	}
+	dir := gitDir(namespace, name, gitURL)
+	headers := map[string]string{}
+	if settings.InstallUUID.Get() != "" {
+		headers["X-Install-Uuid"] = settings.InstallUUID.Get()
+	}
+	// convert caBundle to PEM format because git requires correct line breaks, header and footer.
+	if len(caBundle) > 0 {
+		caBundle = convertDERToPEM(caBundle)
+		insecureSkipTLS = false
+	}
+	return newGit(dir, gitURL, &Options{
+		Credential:        secret,
+		Headers:           headers,
+		InsecureTLSVerify: insecureSkipTLS,
+		CABundle:          caBundle,
+	})
+}
+
+// TODO: Document this properly
+func newGit(directory, url string, opts *Options) (*git, error) {
+	if opts == nil {
+		opts = &Options{}
+	}
+
+	g := &git{
+		URL:               url,
+		Directory:         directory,
+		caBundle:          opts.CABundle,
+		insecureTLSVerify: opts.InsecureTLSVerify,
+		secret:            opts.Credential,
+		headers:           opts.Headers,
+	}
+
+	err := g.setCredential(opts.Credential)
+	if err != nil {
+		return g, err
+	}
+
+	return g, nil
+}
+
+// setCredential sets the authentication credential for the git object.
+// It uses the provided Kubernetes Secret, which contains the authentication information.
+// If the Secret is of type BasicAuth, it extracts the username and password from the Secret,
+// and updates the URL of the git object to include the username, and sets the password field in the git object.
+// If the Secret is of type SSHAuth, it parses the private SSH key from the Secret,
+// adds it to a new SSH agent, and sets the agent in the git object.
+// It also extracts any known hosts from the Secret and sets them in the git object.
+// It returns an error if there are issues parsing the URL or the private SSH key, or adding the key to the SSH agent.
+func (g *git) setCredential(cred *corev1.Secret) error {
+	if cred == nil {
+		return nil
+	}
+
+	if cred.Type == corev1.SecretTypeBasicAuth {
+		username, password := cred.Data[corev1.BasicAuthUsernameKey], cred.Data[corev1.BasicAuthPasswordKey]
+		if len(password) == 0 && len(username) == 0 {
```

Comment:
Both password and username are required? If yes, this should be using OR instead of AND

Reply:
This part I just re-used from the old code, but I agree with you, will change it